src/OrbitGl/TrackManager.cpp

```diff
@@ -380,14 +378,13 @@ bool TrackManager::GetTrackTypeVisibility(Track::Type type) const {
 
 SchedulerTrack* TrackManager::GetOrCreateSchedulerTrack() {
   std::lock_guard<std::recursive_mutex> lock(mutex_);
-  std::shared_ptr<SchedulerTrack> track = scheduler_track_;
-  if (track == nullptr) {
-    track = std::make_shared<SchedulerTrack>(time_graph_, time_graph_, viewport_, layout_, app_,
-                                             capture_data_);
-    AddTrack(track);
-    scheduler_track_ = track;
+  if (scheduler_track_ == nullptr) {
+    auto [unused, track_data] = capture_data_->CreateTrackData();
```

Comment:
Is returning a pair really useful? Should we set the id on the TrackData itself instead?

Reply:
this will become relevant in next PR. The id will be used in new implementation of tracks.

Reply:
to elaborate a bit more - in new tracks we will use id to query the data, TrackData will not directly be used